feature-libs/product-configurator/rulebased/components/price/configurator-price.component.spec.ts

```diff
@@ -89,45 +89,90 @@ describe('ConfiguratorPriceComponent', () => {
       CommonConfiguratorTestUtilsService.expectElementPresent(
         expect,
         htmlElem,
-        '.cx-price'
+        '.cx-greyed-out'
       );
       CommonConfiguratorTestUtilsService.expectElementNotPresent(
         expect,
         htmlElem,
         '.cx-quantity-price'
       );
 
-      expect(component.price).toEqual('+ $10');
+      expect(component.price).toEqual('+$10');
     });
 
-    it('should be defined: value price greater than zero and value is selected', () => {
-      component.formula = createTestData(0, 10, 10, true);
+    it('should be defined: negative value price is shown correctly', () => {
+      component.formula = createTestData(0, -10, 0, false, true);
+      component.formula = {
+        quantity: 0,
+        price: {
+          currencyIso: '$',
+          formattedValue: '-$10',
+          value: -10,
+        },
+        priceTotal: undefined,
+        isLightedUp: true,
+      };
       fixture.detectChanges();
 
       CommonConfiguratorTestUtilsService.expectElementPresent(
         expect,
         htmlElem,
-        '.cx-price'
+        '.cx-greyed-out'
       );
       CommonConfiguratorTestUtilsService.expectElementNotPresent(
         expect,
         htmlElem,
         '.cx-quantity-price'
       );
 
-      expect(component.price).toEqual('+ $10');
+      expect(component.price).toEqual('-$10');
     });
+    it('should be defined: no value price is given and a String of length 0 should be returned', () => {
+      component.formula = createTestData(0, 0, 0, false, true);
```

Comment:
You have to refactor `createTestData` method

Reply:
done